src/main/java/com/checkmarx/intellij/inspections/AscaInspection.java

```diff
@@ -144,7 +144,11 @@ private String escapeHtml(String text) {
     private TextRange getTextRangeForLine(Document document, int lineNumber) {
         int startOffset = document.getLineStartOffset(lineNumber - 1);
         int endOffset = Math.min(document.getLineEndOffset(lineNumber - 1), document.getTextLength());
-        return new TextRange(startOffset, endOffset);
+
+        String lineText = document.getText(new TextRange(startOffset, endOffset));
+        int trimmedStartOffset = startOffset + (lineText.length() - lineText.stripLeading().length());
```

Comment:
what is the behavior of stripLeading() if we have an empty line?

Reply:
Returns a string whose value is this string, with all leading white space removed.
If this String object represents an empty string, or if all code points in this string are white space, then an empty string is returned.
Otherwise, returns a substring of this string beginning with the first code point that is not a white space up to to and including the last code point of this string.
This method may be used to trim white space from the beginning of a string